library/src/validations/maxBytes/maxBytes.ts

```diff
@@ -0,0 +1,31 @@
+import { ValiError } from '../../error';
+import type { ValidateInfo } from '../../types';
+
+/**
+ * Creates a validation functions that validates the byte length of a string.
+ *
+ * @param requirement The maximum length in byte.
+ * @param error The error message.
+ *
+ * @returns A validation function.
+ */
+export function maxBytes<TInput extends string>(
+  requirement: number,
+  error?: string
+) {
+  return (input: TInput, info: ValidateInfo) => {
+    const bytes = encodeURI(input).replace(/%../g, 'x').length;
```

Comment:
Is it true that `encodeURI(input).replace(/%../g, 'x').length` works reliably to calculate the bytes? An alternative could be e.g. `new TextEncoder().encode(input).length`. Furthermore, to reduce the bundle size and since there is only one use, I would not use a variable and add the calculation directly to the if statement.

Reply:
This is an old idiom for calculating the byte length of string in the cross-browser way.
I agree with you `new TextEncoder().encode(input).length` is cleaner. But some of old browser doesn't recognize it.